arpa-exporter/src/worker.py

```diff
@@ -76,7 +76,8 @@ def build_zip(fh: typing.BinaryIO, source_uploads: typing.Iterator[UploadInfo])
     with zipfile.ZipFile(fh, "a") as archive:
         for upload in source_uploads:
             files_checked += 1
-            source_path = os.path.join(DATA_DIR, f"{upload.upload_id}.xlsm")
+            file_extension = upload.path_in_zip.split(".")[-1]
+            source_path = os.path.join(DATA_DIR, f"{upload.upload_id}.{file_extension}")
```

Comment:
This will handle odd edge cases (no file extension, dots in directory names, etc.) slightly better:
```suggestion
            _, file_extension = os.path.splitext(upload.path_in_zip)
            source_path = os.path.join(DATA_DIR, f"{upload.upload_id}{file_extension}")
```

Note: The second value in the 2-tuple returned by `os.path.splitext()` contains the dot (assuming there is one) before the extension (e.g. `os.path.splitext("something.txt") -> ("something", ".txt")`), so important to leave the `.` separator out of the f-string literal.